src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1163,6 +1153,32 @@ private unsafe StringBuilder AppendJoinCore<T>(char* separator, int separatorLen
             return this;
         }
 
+        private unsafe StringBuilder AppendJoinCore<T>(char* separator, int separatorLength, T[] values)
+        {
+            if (values == null)
+                throw new ArgumentNullException(nameof(values));
```

Comment:
@danmosemsft I'd have to add a `values` field to the ExceptionArgument enum, right? https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/ThrowHelper.cs#L342